### DISCUSSION TITLE
Add customisable key bindings to toggle mods

### DISCUSSION THREAD

**Discussion by RoARene317** [ppy/osu#13054] [Ideas]
**Describe the missing feature:** 
In stable release , people could trigger the mods button by pressing keyboard shortcut. Now in lazer it's missing.
**Proposal designs of the feature:*
Mods can be toggle by keyboard , for the custom mods (Grow mods , Transform mods) Can be triggered by pressing F1-F12 (By Default).

**Comment by LanTenDragon** [comment]
If you mean the functionality of pressing A to activate Hard Rock, D for Double Time/Nightcore etc.. that has been implemented already. 

What version of lazer are you running?

And for your suggestion of triggering custom mods with F1-F12, keep in mind that those keys are already assigned to other functions. For example, F1 is assigned to toggle the mods selection, F4 to toggle mute. So you'll need modifier keys + F1-F12.

**Comment by RoARene317** [comment]
![image](https://user-images.githubusercontent.com/24410807/64904897-fe696980-d6fa-11e9-94f4-421dc763af18.png)
2019.913.0 Since this comment is submitted and still not shown

**Comment by frenzibyte** [comment]
@HerendraTJ It's hardcoded to the ruleset itself I think, you can't modify the key binding

**Comment by RoARene317** [comment]
> @HerendraTJ It's hardcoded to the ruleset itself I think, you can't modify the key binding

I think maybe the next update they should add to the key binding since it's a lot of easier than editing the config file.

**Comment by peppy** [comment]
the key bindings on the mod select screen are not intended to be customisable. we may add special mod selection hotkeys which can be used globally, but that is a different scope.
